test/e2e/ue/post-comment.e2e-spec.ts

```diff
@@ -0,0 +1,152 @@
+import {
+  createUser,
+  suite,
+  createUE,
+  makeUserJoinUE,
+  JsonLike,
+} from '../../test_utils';
+import * as pactum from 'pactum';
+import { ERROR_CODE } from '../../../src/exceptions';
+
+const PostCommment = suite('POST /ue/{ueCode}/comments', (app) => {
+  const user = createUser(app);
+  const user2 = createUser(app, { login: 'user2' });
+  const user3 = createUser(app, { login: 'user3' });
+  const ue = createUE(app);
+  makeUserJoinUE(app, user2, ue);
+  makeUserJoinUE(app, user3, ue);
+
+  it('should return a 401 as user is not authenticated', () => {
+    return pactum
+      .spec()
+      .post('/ue/XX00/comments')
+      .withBody({
+        body: 'Test comment',
+      })
+      .expectAppError(ERROR_CODE.NOT_LOGGED_IN);
+  });
+
+  it('should return a 400 because body is a string', () => {
+    return pactum
+      .spec()
+      .withBearerToken(user.token)
+      .post('/ue/XX00/comments')
+      .withBody({
+        body: false,
+        isAnonymous: true,
+      })
+      .expectAppError(ERROR_CODE.PARAM_NOT_STRING, 'body');
+  });
+
+  it('should return a 403 because user has not done the UE yet', () => {
+    return pactum
+      .spec()
+      .withBearerToken(user.token)
+      .post('/ue/XX00/comments')
+      .withBody({
+        body: 'Cette  UE est troooop bien',
+        isAnonymous: true,
+      })
+      .expectAppError(ERROR_CODE.NOT_ALREADY_DONE_UE);
+  });
+
+  it('should return a post a comment in the UE as anonymous user', () => {
+    return pactum
+      .spec()
+      .withBearerToken(user2.token)
+      .post('/ue/XX00/comments')
+      .withBody({
+        body: 'Cette  UE est troooop bien',
+        isAnonymous: true,
+      })
+      .expectUEComment(
+        {
+          id: JsonLike.ANY_UUID,
+          author: {
+            id: user2.id,
+            firstName: user2.firstName,
+            lastName: user2.lastName,
+            studentId: user2.studentId,
+          },
+          createdAt: JsonLike.ANY_DATE,
+          updatedAt: JsonLike.ANY_DATE,
+          semester: {
+            code: 'A24',
+          },
+          isAnonymous: true,
+          body: 'Cette  UE est troooop bien',
+          answers: [],
+          upvotes: 0,
+          upvoted: false,
+        },
+        true,
+      );
+  });
+
+  it('should return a 403 while trying to post another comment', () => {
```

Comment:
limite on peut merge ça avec le test du dessus, et supprimer le commentaire à la fin du test (genre un test "should return a post a comment in the UE as anonymous user and then a 403 when trying to post a second comment") ? ça permettrait de garder l'état initial de la db, et donc de keep track facilement de l'état actuel de la db